src/components/button-link/index.tsx

```diff
@@ -21,14 +27,37 @@ export default function ButtonLink({
   href,
   text,
   ariaLabel,
+  icon,
+  iconPosition = 'leading',
 }: ButtonLinkProps) {
+  const hasIcon = !!icon
+  const hasText = !!text
+  const hasLabel = !!ariaLabel
+  const hasLeadingIcon = hasIcon && iconPosition === 'leading'
+  const hasTrailingIcon = hasIcon && iconPosition === 'trailing'
+  const isIconOnly = hasIcon && !hasText
+
+  if (!hasIcon && !hasText) {
+    throw new Error(
+      '`Button` must have either `text` or an `icon` with accessible labels.'
+    )
+  }
+
+  if (isIconOnly && !hasLabel) {
+    throw new Error(
+      'Icon-only `Button`s require an accessible label. Either provide the `text` prop, or `ariaLabel`.'
+    )
+  }
```

Comment:
Can these messages be updated to reference `ButtonLink` instead of `Button`?

Reply:
✅ Ah great catch, updated!